db/db_impl/db_impl_open.cc

```diff
@@ -475,6 +475,12 @@ Status DBImpl::Recover(
     assert(!files_in_dbname.empty());
     s = versions_->TryRecover(column_families, read_only, files_in_dbname,
                               &db_id_, &missing_table_file);
+#ifndef ROCKSDB_LITE
+    // TODO: remove the VerifyFileChecksums() call because it's very expensive.
+    if (s.ok() && immutable_db_options_.file_checksum_gen_factory) {
```

Comment:
This seems quite general. I would expect some external users to use file_checksum_gen_factory. What if we used a weak symbol (see e.g. jemalloc_helper.h) as an unofficial temporary hook (callback) here?

Reply:
IMO, this line should not be here if I were not trying to create a patch for 6.14.fb. This operation is very expensive and will be removed from here asap after we cherrypick to 6.14

Reply:
This will be executed only if people set `options.best_efforts_recovery` to true.
Not sure if I follow the callback thing. What should we use a callback for?

Reply:
OK, I missed that bit of context, but still, from just our internal customer perspective, it seems very dangerous not to have any way to disable scanning the whole DB on startup other than through disabling best_efforts_recovery.

What I was envisioning was

    extern bool EnableVerifyFileChecksumsOnRecovery() __attribute__((__weak__));
    ...
    if (s.ok() && EnableVerifyFileChecksumsOnRecovery && EnableVerifyFileChecksumsOnRecovery()) {
      ...

Reply:
Valid concern. I am actually more concerned with external users who have to stick with 6.14 but want to set `options.best_efforts_recovery` to `true`. For internal customer, I think they are ok with performing this check by default.

Reply:
Done with a weak symbol `RocksDbFileChecksumsVerificationEnabledOnRecovery`.